### ISSUE TITLE
Newly registered accounts via admin login page can enter admin without being part of shop manager group

### ISSUE THREAD

**Issue by manueldelreal** [reactioncommerce/reaction-admin#292]
## Issue Description
Any account registered via the admin will be automatically logged into the admin, requests for actual information on each section will be denied by the api since the user does not belong to the shop manage group.

## Steps to Reproduce
Please provide starting context, i.e. logged in as a user, configure a particular payment method.
1. Go to localhost:4080
2. If there's an active session, log out
3. If there's no active session, click `register`
4. Enter email and password for an account that is not on the shop manager or owner groups

## Actual results
User will be logged into the admin UI

## Expected results
Unless the user email was previously added to the shop manager or owner groups, the new user should not be able to see the admin UI.

## Possible Solution
It seems that there's a missing permission check when logging into the admin interface.

### Versions
admin: release-v3.0.0-beta.8
reaction: (latest trunk as of 7/13/2020)

**Comment by loan-laux** [comment]
I even think we should disable self-service user sign-up, unless an owner enables a setting to allow it. In this case, there should be a default group assigned to people who sign up on their own. This, too, should be configurable by owners.

Thoughts on this?

**Comment by janus-reith** [comment]
> I even think we should disable self-service user sign-up, unless an owner enables a setting to allow it. In this case, there should be a default group assigned to people who sign up on their own. This, too, should be configurable by owners.
> 
> Thoughts on this?

The IDP is shared for both admin and storefront as you know, so technically, preventing signups is not possible I believe.
What would work, would be to pass some custom query parameter when initiating the oauth flow from the admin to conditionally not render a signin option in the UI.

Still, storefront users could sign in there, not sure if separating this for the two different oauth clients admin/storefront is possible. The api could be used to query the viewer permissions (Which we lacked graphql field(s) to determine that) and based on that redirect or logout right away, this is how I handled it so far in custom solutions.
To not even get to that point, one option could be to give more context to the Meteor signin, which currenlty only has the minimum info that is necessary, as the meteor users dont contain the roles or group logic anymore.
On the server-side part of the Meteor signin, we could fetch the graphql api for the current user to get their permissions, and based on that deny signin if it was intended for the admin ui.

Technically I could still spoof that by using a separate storefront-intended Meteor signin flow and interrupting that before the automatic signout that happens automatically in the IDP before the hydra redirect, and reuse them on some admin logic, so while no real threat, that could be prevented by maybe adding that permission check to both the Meteor signin and the `WebApp` handler the IDP exposes to redirect back to hydra and complete the flow.
